packages/presentation/src/loader/BroadcastDisplayedDocument.tsx

```diff
@@ -1,20 +1,62 @@
 import {memo, useEffect} from 'react'
 import type {SanityDocument} from 'sanity'
+import {useDocumentPane} from 'sanity/structure'
 
-import {useDisplayedDocumentBroadcaster} from './DisplayedDocumentBroadcaster'
+import type {PresentationParams} from '../types'
+import {usePresentationParams} from '../usePresentationParams'
+import {
+  type SetDisplayedDocument,
+  useDisplayedDocumentBroadcaster,
+} from './DisplayedDocumentBroadcaster'
 
 /**
  * Sanity Form input component that reads the current form state and broadcasts it to
  * the live query store
  */
 function BroadcastDisplayedDocument(props: {
   value: Partial<SanityDocument> | null | undefined
-}): null {
+}): JSX.Element | null {
   const setDisplayedDocument = useDisplayedDocumentBroadcaster()
+  const params = usePresentationParams(false)
 
-  useEffect(() => setDisplayedDocument?.(props.value), [props.value, setDisplayedDocument])
+  useEffect(() => {
+    if (params?.perspective !== 'published') {
+      const timeout = setTimeout(() => setDisplayedDocument?.(props.value), 10)
```

Comment:
Is there any sort of deeper meaning behind the 10ms timeout here and the 100ms timeout below?

Reply:
Well, technically we could remove the 10ms all together on this one. The timeout is only needed for the published document lookup as it takes a moment to load the `editState` :)